src/components/Layout.tsx

```diff
@@ -105,6 +130,11 @@ export function Layout() {
         </select>
       </div>
 
+      <div className="sort-row">
+        <input type="submit" value="Filter rules" onClick={() => setShowRules(!showRules)} />
+        { showRules ? <Rules state={matchOption} setState={setMatchOption} /> : null }
```

Comment:
This could be handled with a HTML `<details>` element. e.g.

```html
<details>
	<summary>Show rules</summary>
	<Rules state={matchOption} setState={setMatchOption} />
</details>
```

demo:

<details>
	<summary>Show rules</summary>
	<p>rules go here
</details>

Reply:
Well, that's a lot simpler for sure!